functions/api/routes/kpi.js

```diff
@@ -194,4 +196,156 @@ router.get('/:id/values', param('id').trim().escape(), async (req, res) => {
   }
 });
 
+router.put(
+  '/:id/values/:date',
+  teamSecretValidator,
+  idValidator,
+  dateValidator,
+  valueValidator,
+  async (req, res) => {
+    const result = validationResult(req);
+
+    if (!result.isEmpty()) {
+      res.status(400).json({
+        message: 'Invalid request data',
+        errors: result.mapped(),
+      });
+      return;
+    }
+
+    const { 'okr-team-secret': teamSecret, id, date, value } = matchedData(req);
+
+    const db = getFirestore();
+
+    try {
+      const kpi = await db.collection('kpis').doc(id).get();
+
+      const { exists, ref } = kpi;
+
+      if (!exists) {
+        res.status(404).json({ message: `Could not find KPI with ID: ${id}` });
+        return;
+      }
+
+      const { parent } = kpi.data();
+      const parentData = await parent.get().then((snapshot) => snapshot.data());
+
+      if (!parentData.secret) {
+        res.status(401).json({
+          message:
+            `'${parentData.name}' is not set up for API usage. Please set ` +
+            'a secret using the OKR Tracker admin interface.',
+        });
+        return;
+      }
+      if (parentData.secret !== teamSecret) {
+        res.status(401).json({ message: 'Wrong okr-team-secret' });
+        return;
+      }
+
+      await updateKPIProgressionValue(ref, date, value);
+
+      res.json({
+        message: `KPI progression value for ${format(
+          date,
+          'yyyy-MM-dd'
+        )} updated to ${value}`,
```

Comment:
Kanskje trekke ut `format`-uttrykket for bedre lesbarhet?

Reply:
Hvor bør vi ha disse? Kanskje mer ideelt å gjenbruke noe fra `src/util/format.js`, men vet ikke om det da bør flyttes opp et nivå til `common`? 🤔

Reply:
Nei tenkte bare rent syntaktisk. :P Å trekke det ut i en variabel på linja over for eksempel. Men smak og behag og ikke viktig det heller. :sweat_smile:

Reply:
Ah 😓